src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,3047 @@
+                batch::inner(pay(alice, bob, XRP(1)), 0, aliceTicketSeq + 1),
+                ticket::use(aliceTicketSeq));
+            auto const objTxn = env.jt(check::cash(bob, chkId, XRP(10)));
+            env(objTxn, ter(tecNO_ENTRY));
+            env(batchTxn);
+            env.close();
+
+            auto const txIDs = batchTxn.stx->getBatchTransactionIDs();
+            TxID const parentBatchId = batchTxn.stx->getTransactionID();
+            Json::Value const jrr = getLastLedger(env);
+            std::vector<TestLedgerData> testCases = {
+                {0, "Batch", "tesSUCCESS", to_string(parentBatchId), false},
+                {1, "CheckCreate", "tesSUCCESS", to_string(txIDs[0]), true},
+                {2, "Payment", "tesSUCCESS", to_string(txIDs[1]), true},
+                {3,
+                 "CheckCash",
+                 "tesSUCCESS",
+                 to_string(objTxn.stx->getTransactionID()),
+                 false},
+            };
+            validateLedgerTxns(env, jrr, testCases, parentBatchId);
+        }
+
+        // After Batch Txn
+        {
+            test::jtx::Env env{*this, envconfig()};
+            env.fund(XRP(1000), alice, bob);
+            env.close();
+
+            std::uint32_t aliceTicketSeq{env.seq(alice) + 1};
+            env(ticket::create(alice, 10));
+            env.close();
+
+            auto const aliceSeq = env.seq(alice);
+            auto const batchFee = batch::calcBatchFee(env, 0, 2);
+            uint256 const chkId{getCheckIndex(alice, aliceSeq)};
+            auto const batchTxn = env.jt(
+                batch::outer(alice, 0, batchFee, tfAllOrNothing),
+                batch::inner(check::create(alice, bob, XRP(10)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), 0, aliceTicketSeq + 1),
+                ticket::use(aliceTicketSeq));
+            auto const objTxn = env.jt(check::cash(bob, chkId, XRP(10)));
+            env(batchTxn);
+            env(objTxn, ter(tecNO_ENTRY));
+            env.close();
+
+            auto const txIDs = batchTxn.stx->getBatchTransactionIDs();
+            TxID const parentBatchId = batchTxn.stx->getTransactionID();
+            Json::Value const jrr = getLastLedger(env);
+            std::vector<TestLedgerData> testCases = {
+                {0, "Batch", "tesSUCCESS", to_string(parentBatchId), false},
+                {1, "CheckCreate", "tesSUCCESS", to_string(txIDs[0]), true},
+                {2, "Payment", "tesSUCCESS", to_string(txIDs[1]), true},
+                {3,
+                 "CheckCash",
+                 "tesSUCCESS",
+                 to_string(objTxn.stx->getTransactionID()),
+                 false},
+            };
+            validateLedgerTxns(env, jrr, testCases, parentBatchId);
+        }
+    }
+
+    void
+    testPseudoTxn(FeatureBitset features)
+    {
+        testcase("pseudo txn");
+
+        using namespace test::jtx;
```

Comment:
I'm confused on what this test is testing - shouldn't the pseudo-transaction be included in a Batch?

Reply:
It's testing `tfInnerBatchTxn`. Should I rename it? Or do you want a test testing the transaction as a pseudo transaction?

Reply:
Maybe have some callout in a comment or the testcase name saying that? e.g. `testcase("pseudo txn with tfInnerBatchTxn")`

Reply:
fixed